### PR TITLE
Enable CI builds

### PR DESCRIPTION
**Pull request by karanthukral**
## What?
- Replicates the build pipeline/method being used in shopify-docker-images
- Having a separate repo will make it easier to keep up to date with upstream without having to use submodules.
- Currently pushes to a different location but after review I'll enable automatic builds and pushing to the same location and this will move ingress out of the shopify-docker-images repo

cc/ @ibawt @n1koo

### CODE REVIEW

File: controllers/nginx/Makefile [hunk 1]
```diff
@@ -24,91 +20,20 @@ PKG=k8s.io/ingress/controllers/nginx
 
 ARCH ?= $(shell go env GOARCH)
 GOARCH = ${ARCH}
-DUMB_ARCH = ${ARCH}
```
Comment (dalehamel) [inline controllers/nginx/Makefile]: why did you delete a bunch of the make file? Couldn't you just not call these specific rules

File: controllers/nginx/Makefile [hunk 2]
```diff
@@ -24,91 +20,20 @@ PKG=k8s.io/ingress/controllers/nginx
...
-	DUMB_ARCH=ppc64el
-endif
-#ifeq ($(ARCH),s390x)
-#        QEMUARCH=s390x
-#endif
-
-TEMP_DIR := $(shell mktemp -d)
 
-DOCKERFILE := $(TEMP_DIR)/rootfs/Dockerfile
+container:
```
Comment (dalehamel) [inline controllers/nginx/Makefile]: it looks like the main reason you have this step is to reference the tag

File: controllers/nginx/Makefile [hunk 3]
```diff
@@ -24,91 +20,20 @@ PKG=k8s.io/ingress/controllers/nginx
...
-ifeq ($(ARCH), amd64)
-	# This is for to maintain the backward compatibility
-	$(DOCKER) tag $(MULTI_ARCH_IMG):$(TAG) $(IMAGE):$(TAG)
-endif
-
-push: .push-$(ARCH)
-.push-$(ARCH):
-	$(DOCKER) push $(MULTI_ARCH_IMG):$(TAG)
-ifeq ($(ARCH), amd64)
+push: container
```
Comment (dalehamel) [inline controllers/nginx/Makefile]: when is push called? I don't see it

File: controllers/nginx/rootfs/Dockerfile [hunk 4]
```diff
@@ -12,21 +12,18 @@
...
   && rm -rf /var/lib/apt/lists/*
 
-RUN curl -sSL -o /tmp/dumb-init.deb http://ftp.us.debian.org/debian/pool/main/d/dumb-init/dumb-init_1.2.0-1_DUMB_ARCH.deb && \
-  dpkg -i /tmp/dumb-init.deb && \
-  rm /tmp/dumb-init.deb
+RUN curl -sSL -o /sbin/tini https://github.com/krallin/tini/releases/download/v0.15.0/tini-amd64 && \
+  chmod +x /sbin/tini
 
-ENTRYPOINT ["/usr/bin/dumb-init"]
+ENTRYPOINT ["/sbin/tini", "--"]
```
Comment (dalehamel) [inline controllers/nginx/rootfs/Dockerfile]: is tini just a drop in replacement?

File: controllers/nginx/rootfs/Dockerfile [hunk 5]
```diff
@@ -12,21 +12,18 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM BASEIMAGE
-
-CROSS_BUILD_COPY qemu-QEMUARCH-static /usr/bin/
+FROM gcr.io/google_containers/nginx-slim-amd64:0.23
```
Comment (dalehamel) [inline controllers/nginx/rootfs/Dockerfile]: is this what the baseimage was?

File: controllers/nginx/Makefile (hunk 1 above)
Comment (n1koo) [inline controllers/nginx/Makefile]: ^  agree, easier to keep sync with upstream to not do a lot of changes

File: controllers/nginx/rootfs/Dockerfile (hunk 4 above)
Comment (n1koo) [inline controllers/nginx/rootfs/Dockerfile]: It should be, but do we need it ? Eg. do have a reason to drift from upstream?

File: controllers/nginx/Makefile (hunk 1 above)
Comment (karanthukral) [inline controllers/nginx/Makefile]: I replicated the build pipeline we were using in shopify-docker-images. I'll try use the default makefile and pipa constraints

File: controllers/nginx/Makefile (hunk 3 above)
Comment (karanthukral) [inline controllers/nginx/Makefile]: push isn't called. Thats done by pipa

File: controllers/nginx/rootfs/Dockerfile (hunk 5 above)
Comment (karanthukral) [inline controllers/nginx/rootfs/Dockerfile]: Ya. I was having issues it picking up the `BASEIMAGE` defined in the makefile so I explicitly stated it here

File: controllers/nginx/rootfs/Dockerfile (hunk 4 above)
Comment (karanthukral) [inline controllers/nginx/rootfs/Dockerfile]: I was having issues with using `dumb-init` that's why I switched to the one from shopify-docker-images. I'll see if I can rely on using the default makefile

Comment (n1koo) [inline controllers/nginx/rootfs/Dockerfile]: If you are having issues that can't be easily fixed/worked around thats a legit reason :)

Comment (dalehamel) [inline controllers/nginx/rootfs/Dockerfile]: yep fair enough

**Comment by karanthukral** [comment]
@dalehamel @n1koo Not sure why I ran into issues yesterday but I got the builds working with minimal changes

File: controllers/nginx/Makefile [hunk 6]
```diff
@@ -30,7 +30,7 @@ ALL_ARCH = amd64 arm arm64 ppc64le
 
 QEMUVERSION=v2.9.1
 
-IMGNAME = nginx-ingress-controller
+IMGNAME = nginx-ingress-controller-fork
```
Comment (karanthukral) [inline controllers/nginx/Makefile]: I will change this once we are okay with the PR and remove the ingress from the other repo

**Comment by dalehamel** [comment]
@karanthukral i like this much better :clap:

**Comment by karanthukral** [comment]
> i like this much better 👏

I agree. Might have seen issues cause I was also changing the build pipeline a lot yesterday!
